blog/secadvisor.md

```diff
@@ -0,0 +1,290 @@
+
+Follow the instructions on https://docs.min.io/docs/minio-quickstart-guide.html to install and run a minio object store (an AWS S3-like object store).
+
+```
+wget https://dl.min.io/server/minio/release/linux-amd64/minio
+chmod +x minio
+./minio server /data
+```
+
+This will run a basic minio object store server on your machine, placing the objects in "/data".
+Note the printed output in the logs from the ./minio command and take down the values of "Endpoint", "AccessKey", and "SecretKey".
+Place these items in the secadvisor.yml in the appropriate locations.
+Create a bucket in the object store to hold the security-advisor data.
+Additional details on this option appear in https://github.com/skydive-project/skydive/tree/master/contrib/pipelines/secadvisor.
+
+## Deploy the Security Advisor pipeline
+
+Build and run the pipeline in the secadvisor directory: $SKYDIVE_BASE/contrib/pipelines/secadvisor.
+
+```
+cd $SKYDIVE_BASE/contrib/pipelines/secadvisor
+make static
+```
+
+The binary is created in the same directory: $SKYDIVE_BASE/contrib/pipelines/secadvisor/secadvisor.
+
+### Activate the Security Advisor
+
+```
+secadvisor /etc/skydive/secadvisor.yml
+```
+
+Note that it is possible to run several instances of the Security Advisor at the same time, each one with a different configuration file.
+This is useful to capture different flows for different purposes and to perform different filtering (e.g. based on specified cluster_net_masks and filter parameters).
+
+## Generate and Capture Flows
+
+### Capture Flows
+
+Via the Skydive WebUI setup captures and generate traffic which should
+result in the secadvisor pipeline sending flows to the ObjectStore.
+
+To connect to the GUI, open a web browzer to the address of the Skydive analyzer at port 8082. If running the browzer on the same machine as the analyzer, then connect to localhost:8082.
+You should see the topology of your network, perhaps something like the following image.
+
+<p class="center">
+  <a href="/assets/images/blog/capture-future-1.png" data-lightbox="Skydive Topology" data-title="Skydive topology view">
+    <img src="/assets/images/blog/capture-future-1.png"/>
+  </a>
+</p>
+
+
+Make sure you are on the "Captures" view and press on "Create".
+
+Fill in the "Targets" fields by placing the cursor in the first "Interface" field and then use the mouse to point to your network endpoint that you want to capture.
+To capture a particular flow end-to-end, fill in the second "Interface" field in a simmilar manner with the other network endpoint.
+Then press the "Start" button on the GUI.
+
+Alternatively, you can start the capture of flows from the command line with a commond like the following:
+
+```
+skydive client capture create --gremlin "G.V().Has('Type', 'device', 'Name', 'eth0')" --type pcap 
+```
+
+The command specifies to capture all flows that match the gremlin expression - in this case, all devices that have name 'eth0'.
+
+### Generate Flows
```

Comment:
you can use the inject feature of Skydive rather than introducing a new tool

Reply:
This is mentioned at the end of the section.

Reply:
ok